### ISSUE TITLE
Custom node inheriting BaseButton does not appear in Create New Node dialog

### ISSUE THREAD

**Issue by ret80** [godotengine/godot#8034]
**Operating system or device - Godot version:**
Godot 2.1.x

**Issue description:**
It took me to create my own button. Based on the class of the base button. But I found out that this button is not shown in the dialog "create a new node". If I create my class from a button then my button is added to the list. The problem is that I do not need the functionality of the button. I need the basic button functionality that I want to extend. And this can not be done.
For a better understanding of the problem, I created an example.
In the menu "create new node" should appear 2 buttons created by me. But there is only one: (

plugin.gd
```
tool
extends EditorPlugin

func _enter_tree():
	add_custom_type("MyButton", "Button", preload("myButton.gd"), null)
	add_custom_type("MyBaseButton", "BaseButton", preload("myBaseButton.gd"), null)

func _exit_tree():
	remove_custom_type("MyButton")
	remove_custom_type("MyBaseButton")
```

![image](https://cloud.githubusercontent.com/assets/233268/23940800/5811f9c4-0977-11e7-8651-25e0850b22cd.png)

**Steps to reproduce:**

**Link to minimal example project:**

https://www.dropbox.com/s/py9r5c2lezu9n6b/BaseButton.zip?dl=0

**Comment by Tarocco** [comment]
Godot 2.1 here!
You might try re-activatingthe addon/plugin in the project settings.
Go to
**Scene > Project Settings > Plugins** tab
and set the plugin state to Inactive, then Active, and see if that helps.

Maybe future versions of Godot can reload the addon/plugin script automatically 😄

**Comment by ret80** [comment]
The problem is not at all in this. The base button does not have a constructor. And everything that is inherited from the base button is simply skipped.

**Comment by bojidar-bg** [comment]
As mentioned in #9495, affects all "virtual" classes (Joint2d, etc.), also, affects 3.0/master.

**Comment by kubecz3k** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about its current status here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by DualMatrix** [comment]
This has been fixed in 3.1 thanks to class_name superseding the old custom nodes system.
